File: Longest Repeating Character Replacement - Leetcode 424/Longest Repeating Character Replacement - Leetcode 424.java
public class Solution {
    public int characterReplacement(String s, int k) {
        int[] counts = new int[26];
        int l = 0, longest = 0;
        int maxCount = 0;

        for (int r = 0; r < s.length(); r++) {
            maxCount = Math.max(maxCount, ++counts[s.charAt(r) - 'A']);

            while ((r - l + 1) - maxCount > k) {
                counts[s.charAt(l) - 'A']--;
                l++;
            }

            longest = Math.max(longest, r - l + 1);
        }

        return longest;
    }
}

File: Longest Repeating Character Replacement - Leetcode 424/Longest Repeating Character Replacement - Leetcode 424.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> counts(26, 0);
        int l = 0, longest = 0, maxCount = 0;

        for (int r = 0; r < s.size(); r++) {
            maxCount = max(maxCount, ++counts[s[r] - 'A']);

            while ((r - l + 1) - maxCount > k) {
                counts[s[l] - 'A']--;
                l++;
            }

            longest = max(longest, r - l + 1);
        }

        return longest;
    }
};
